repose-aggregator/core/core-lib/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -374,6 +382,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                 }
                 requestFilterChain.startFilterChain(mutableHttpRequest, mutableHttpResponse);
             }
+        } catch (InvalidMethodException ime) {
+            LOG.debug("{}:{} -- Invalid HTTP method requested: {}", clusterId, nodeId, mutableHttpRequest.getMethod(), ime);
+            mutableHttpResponse.sendError(HttpServletResponse.SC_BAD_REQUEST, "Error processing request");
+            mutableHttpResponse.setLastException(ime);
```

Comment:
Quickly looking at the code, if we used a try/catch earlier, we'd lose all of the work that happens in the finally.

The proposed approach gives us clarity and lets us capture some metrics by making use of the finally block.

Ship it.